java-frontend/src/main/java/org/sonar/java/resolve/WildCardType.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * SonarQube Java
+ * Copyright (C) 2012-2016 SonarSource SA
+ * mailto:contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.java.resolve;
+
+import org.sonar.plugins.java.api.semantic.Type;
+
+public class WildCardType extends JavaType {
+
+  public enum BoundType {
+    UNBOUNDED("?"),
+    SUPER("? super "),
+    EXTENDS("? extends ");
+
+    private final String name;
+
+    BoundType(String name) {
+      this.name = name;
+    }
+
+    @Override
+    public String toString() {
+      return name;
+    }
+  }
+
+  final JavaType bound;
+  final BoundType boundType;
+
+  public WildCardType(JavaType bound, BoundType boundType) {
+    super(WILDCARD, new JavaSymbol.WildcardSymbol(boundType == BoundType.UNBOUNDED ? boundType.toString() : (boundType + bound.symbol.name())));
+    this.bound = bound;
+    this.boundType = boundType;
+    this.symbol.type = this;
+  }
+
+  @Override
+  public boolean isSubtypeOf(String fullyQualifiedName) {
+    return "java.lang.Object".equals(fullyQualifiedName) || (boundType == BoundType.EXTENDS && bound.isSubtypeOf(fullyQualifiedName));
+  }
+
+  @Override
+  public boolean isSubtypeOf(Type superType) {
+    if (((JavaType) superType).isTagged(WILDCARD)) {
+      org.sonar.java.resolve.WildCardType superTypeWildcard = (org.sonar.java.resolve.WildCardType) superType;
+      JavaType superTypeBound = superTypeWildcard.bound;
+      switch (superTypeWildcard.boundType) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Add a default case to this switch. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3ASwitchLastCaseIsDefaultCheck)